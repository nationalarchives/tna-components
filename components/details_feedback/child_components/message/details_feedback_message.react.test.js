import React from 'react';
import Message from './details_feedback_message.react';
import TestRenderer from 'react-test-renderer';
import renderer from 'react-test-renderer';

let testRenderer, testRendererToJson;

// Snapshot 1
it('renders correctly', () => {
  const items = ['Hello message component'];
  const tree = renderer.create(<Message message={items[0]} />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Snapshot 2
it('renders the Message component with optional props', () => {
  const items = ['Hello message component', 'message-class'];
  const tree = renderer
    .create(<Message message={items[0]} className={items[1]} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// Unit test
describe("Check DOM element and it's attributes", () => {
  const items = ['Hello message text'];
  testRenderer = TestRenderer.create(<Message message={items[0]} />);

  testRendererToJson = testRenderer.toJSON();
  it('Should have an <div> tag', () => {
    expect(testRendererToJson.type).toBe('p');
  });
  it('Should have a prop of className', () => {
    expect(testRendererToJson.props.hasOwnProperty('className')).toBeTruthy();
  });
  it('Should have children props', () => {
    expect(testRendererToJson.children.length).toBe(1);
  });
});
