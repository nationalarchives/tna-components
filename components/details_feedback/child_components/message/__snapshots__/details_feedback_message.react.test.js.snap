// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<p
  className={null}
>
  Hello message component
</p>
`;

exports[`renders the Message component with optional props 1`] = `
<p
  className="message-class"
>
  Hello message component
</p>
`;
